services/templeton/src/templeton/database/database.go

```diff
@@ -0,0 +1,200 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"logcabin"
+
+	"templeton/model"
+
+	_ "github.com/lib/pq"
+)
+
+var (
+	EOS    = errors.New("EOS")
+	logger = logcabin.New()
+)
+
+// Databaser is a type used to interact with the database.
+type Databaser struct {
+	db         *sql.DB
+	ConnString string
+}
+
+// NewDatabaser returns a pointer to a Databaser instance that has already
+// connected to the database by calling Ping().
+func NewDatabaser(connString string) (*Databaser, error) {
+	db, err := sql.Open("postgres", connString)
+	if err != nil {
+		return nil, err
+	}
+	err = db.Ping()
+	if err != nil {
+		return nil, err
+	}
+	databaser := &Databaser{
+		db:         db,
+		ConnString: connString,
+	}
+	return databaser, nil
+}
+
```

Comment:
The stuff above here is pretty much copied from jobservices/jex-go, and the stuff below here is new in this service.